src/Language/Bitcoin/Script/Descriptors/Parser.hs

```diff
@@ -41,7 +46,16 @@ parseDescriptor :: Network -> Text -> Either String OutputDescriptor
 parseDescriptor net = A.parseOnly $ outputDescriptorParser net
 
 outputDescriptorParser :: Network -> Parser OutputDescriptor
```

Comment:
How about having a more structured parse result?:

```haskell
data ChecksumStatus = Valid | Invalid | Absent

data ChecksumDescriptor = ChecksumDescriptor 
  { descriptor :: OutputDescriptor
  , checksumStatus :: ChecksumStatus
  }
```

That way the application logic can handle the various possibilities without having to choose between parsing logic that does or does-not require the checksum.

Reply:
Nice idea. I've added a similar type to the one you suggested in the syntax module and added a parser for it.